src/containers/Cinema/Cinema.tsx

```diff
@@ -1,192 +1,139 @@
-    playerRef.current?.off('play');
-    playerRef.current?.off('pause');
-    playerRef.current?.off('userActive');
-    playerRef.current?.off('userInactive');
-  }, []);
-
-  useEffect(() => {
-    if (!window.jwplayer && !loadingRef.current) {
-      loadingRef.current = true;
-
-      addScript(scriptUrl).then(() => {
-        setLibLoaded(true);
-        loadingRef.current = false;
-      });
-    }
-  }, [scriptUrl]);
-
-  useEffect(() => {
-    if (!player) {
-      return;
-    }
-
-    const calculateWatchHistoryProgress = () => {
-      const { watchHistory } = useWatchHistoryStore.getState();
-      const watchHistoryItem = watchHistory.find(({ mediaid }) => mediaid === item.mediaid);
-
-      if (
-        watchHistoryItem &&
-        enableWatchHistory &&
-        watchHistoryItem.progress > VideoProgressMinMax.Min &&
-        watchHistoryItem.progress < VideoProgressMinMax.Max
-      ) {
-        seekToRef.current = watchHistoryItem.progress * watchHistoryItem.duration;
-      } else {
-        seekToRef.current = -1;
-      }
-    };
-
-    const loadPlaylist = () => {
-      if (!item || !playerRef.current) {
-        return;
-      }
-
-      const currentItem = playerRef.current?.getPlaylistItem() as PlaylistItem | null;
-
-      // we already loaded this item
-      if (currentItem && currentItem.mediaid === item.mediaid) {
-        return;
-      }
-      // load new item
-      playerRef.current.load([deepCopy(item)]);
-      calculateWatchHistoryProgress();
-    };
-
-    const initializePlayer = () => {
-      if (!window.jwplayer || !playerElementRef.current) return;
-
-      playerRef.current = window.jwplayer(playerElementRef.current) as JWPlayer;
-
-      playerRef.current.setup({
-        playlist: [deepCopy(item)],
-        aspect: false,
-        width: '100%',
-        height: '100%',
-        mute: false,
-        autostart: true,
-        repeat: false,
-      });
-
-      attachEvents();
-      calculateWatchHistoryProgress();
-      setPlayer(playerRef.current);
-
-      playerRef.current.setPlaylistItemCallback(handlePlaylistItemCallback);
-    };
-
-    if (playerRef.current) {
-      return loadPlaylist();
-    }
-
-    if (libLoaded) {
-      initializePlayer();
-    }
-  }, [libLoaded, item, enableWatchHistory, setPlayer, handlePlaylistItemCallback, detachEvents, attachEvents, player]);
+  const handleUserActive = useCallback(() => setUserActive(true), []);
+  const handleUserInactive = useCallback(() => setUserActive(false), []);
+  const handlePlaylistItemCallback = usePlaylistItemCallback();
 
+  // effects
   useEffect(() => {
-    return () => {
-      if (playerRef.current) {
-        // Detaching events before component unmount
-        detachEvents();
-        playerRef.current.remove();
-      }
-    };
-  }, [detachEvents]);
+    if (open) setUserActive(true);
+    if (!open) saveItem(item, getProgress());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This is needed since we only want this effect to run when the `open` property updates

Reply:
Will `getProgress` or `item` change though?  I think they shouldn't.  If you end up leaving the `disable-next-line`, just add the comment ("This is needed since we only want this effect to run when the open property updates") with it

Reply:
`item` will change when the next video starts.
`item` is a dependency of `getProgress`, so that will change as well.

I will add a comment 👍